### PR TITLE
Docs: Request

### PR DESCRIPTION
**Pull request by sam-finch-tezos**
# Context

**Related Tasks**: [API doc: Request](https://app.asana.com/0/1205170694555856/1205457666696783/f)

# Description

This PR an API reference and quick start for the `Request` runtime API

# Manually testing the PR
```sh
npm run docs:dev
```

### CODE REVIEW

File: docs/api/request.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# 🙏 Request
+
+`jstz`'s [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) implementation is based on the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) specification. This API permits you to manipulate and inspect HTTP request and response headers.
```
Comment (johnyob) [inline docs/api/request.md:3]: 'This API permits you to manipulate and inspect HTTP requests'

File: docs/api/request.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+::: danger
+⚠️ `jstz`'s implementation is not spec compliant ⚠️
+:::
+
+## Constructors
+### `new Request(input : Request)`
+Constructs a new request object as a copy of its argument.
+### `new Request(input : string | URL, options: RequestOptions)`
+Creates a new `Request` object.
+The first argument may be either a string url, or a [`URL`](./url.md) object.
```
Comment (johnyob) [inline docs/api/request.md:14]: Not too sure we support a `URL` as input.

File: docs/api/request.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+`jstz`'s [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) implementation is based on the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) specification. This API permits you to manipulate and inspect HTTP request and response headers.
+
+::: danger
+⚠️ `jstz`'s implementation is not spec compliant ⚠️
+:::
+
+## Constructors
+### `new Request(input : Request)`
+Constructs a new request object as a copy of its argument.
+### `new Request(input : string | URL, options: RequestOptions)`
```
Comment (johnyob) [inline docs/api/request.md:12]:
```suggestion
### `new Request(input : string, init?: RequestInit)`
```

File: docs/api/request.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  | {
+      method?: 'GET' | 'CONNECT' | 'TRACE' | 'OPTIONS' | 'HEAD' | 'DELETE';
+      headers?: Headers | { [headerName: string]: string };
+    }
+  | {
+      method: 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+      headers?: Headers | { [headerName: string]: string };
+      body: ArrayBuffer | string;
+    };
+```
```
Comment (johnyob) [inline docs/api/request.md:27]: Please use the type from our TypeScript SDK
```
declare interface RequestInit {
  body?: BodyInit | null;
  headers?: HeadersInit;
  method?: string;
}

File: docs/api/request.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+⚠️ `jstz`'s implementation is not spec compliant ⚠️
+:::
+
+## Constructors
+### `new Request(input : Request)`
+Constructs a new request object as a copy of its argument.
+### `new Request(input : string | URL, options: RequestOptions)`
+Creates a new `Request` object.
+The first argument may be either a string url, or a [`URL`](./url.md) object.
+The second, optional, argument is a javascript object containing any custom settings for the request.
```
Comment (johnyob) [inline docs/api/request.md:15]:
```suggestion
The second, optional, argument is a JavaScript object containing any custom settings for the request.
```

File: docs/api/request.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+The available settings are.
+#### `method`
+A string representing the `http` method of the request, eg `'GET'`, `'PUT'`, `'POST'`. 
+If omitted the default value is `'GET'`.
+#### `headers`
+Any headers that should be attatched to the request.
+Either a [`Headers`](./headers.md) object or an object literal whos values are strings.
+#### `body`
+The body attached to the request. 
+This can either be a string or a javascript array buffer.
```
Comment (johnyob) [inline docs/api/request.md:37]: JavaScript

Comment (johnyob) [reply docs/api/request.md:37]: Additionally please put a spec deviation warning here for the unsupported options + differences in `body`

File: docs/api/request.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+`'GET'`, `'CONNECT'`, `'TRACE'`, `'OPTIONS'` and `'HEAD'` methods.
+
+## Instance Properties
+### `readonly bodyUsed: bool`
+A boolean value representing whether the body has been accessed.
+### `readonly headers`
+A `Headers` object containing the headers attached to the request
+### `readonly method: string`
+A string representing the `http` method of the request, eg `'GET'`, `'PUT'`, `'POST'`. 
+### `readonly url: URL`
```
Comment (johnyob) [inline docs/api/request.md:48]:
```suggestion
### `readonly url: string`
```

File: docs/api/request.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+The body is required for the `'PUT'`, `'POST'` and `'PATCH'` methods and forbidden for the 
+`'GET'`, `'CONNECT'`, `'TRACE'`, `'OPTIONS'` and `'HEAD'` methods.
+
+## Instance Properties
+### `readonly bodyUsed: bool`
+A boolean value representing whether the body has been accessed.
+### `readonly headers`
+A `Headers` object containing the headers attached to the request
+### `readonly method: string`
+A string representing the `http` method of the request, eg `'GET'`, `'PUT'`, `'POST'`.
```
Comment (johnyob) [inline docs/api/request.md:47]: `http` -> HTTP

File: docs/api/request.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+A `URL` object containing the request url.
+
+## Instance Methods
+### `async arrayBuffer() : Promise<ArrayBuffer>`
+Returns a promise containing the response body as an
+[`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/Request/arrayBuffer).
+
+### `async json() : Promise<unknown>`
+Returns a promise containing the response body parsed as JSON.
+### `async text() : Promise<String>`
```
Comment (johnyob) [inline docs/api/request.md:58]:
```suggestion
### `async text() : Promise<string>`
```

File: docs/api/request.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+Returns a promise containing the response body as a string.
```
Comment (johnyob) [inline docs/api/request.md:59]: Mention encoding (UTF-16)

File: docs/api/request.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+A string representing the `http` method of the request, eg `'GET'`, `'PUT'`, `'POST'`. 
+### `readonly url: URL`
+A `URL` object containing the request url.
+
+## Instance Methods
+### `async arrayBuffer() : Promise<ArrayBuffer>`
+Returns a promise containing the response body as an
+[`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/Request/arrayBuffer).
+
+### `async json() : Promise<unknown>`
```
Comment (johnyob) [inline docs/api/request.md:56]:
```suggestion
### `async json() : Promise<any>`
```

**Review by asutosh-mourya — COMMENTED** [review]
the document is pretty well reviewed already please address the comments, Nothing else to add from my side.

**Comment by johnyob** [comment]
@sam-finch-tezos please see [a8fca08](https://github.com/trilitech/jstz/pull/56/commits/a8fca082bcab87615302ffc68b76325d70c48cb2) for the review commit addressing all comments above

**Pull request merged**